src/commands/kv/mod.rs

```diff
@@ -19,15 +20,7 @@ pub mod namespace;
 // Create a special API client that has a longer timeout than usual, given that KV operations
 // can be lengthy if payloads are large.
 fn api_client(user: &GlobalUser) -> Result<HttpApiClient, failure::Error> {
-    http::cf_v4_api_client(
-        user,
-        HttpApiClientConfig {
-            default_headers: http::headers(None),
-            // Use 5 minute timeout instead of default 30-second one.
-            // This is useful for bulk upload operations.
-            http_timeout: Duration::from_secs(5 * 60),
-        },
-    )
+    http::cf_v4_api_client(user, Some(Feature::KV))
```

Comment:
these are trying to do vastly different things and i don't think this is the right move.

Reply:
👍 i've updated the `cf_api_client` to take a `CfApiClientConfig` struct which has a builder API and a default

Reply:
this allows you to either pass `CfApiClientConfig::default()` as the second parameter or you could do `CfApiClientConfig::builder().timeout(Duration::from_secs(5 * 60)).feature(Feature::Sites).build()`

I've left the legacy api without something similar because we dont need to overengineer something we're moving away from